### ISSUE TITLE
allow routes to operate on location hash

### ISSUE THREAD

**Issue by kingsleyh** [mint-lang/mint#101]
At the moment I can't run Kemal and server the API and the Mint app from the same kemal instance. This means I have to run 2 servers and I can't make use of Kemals session handling and I have to enable cors. 

Because the routes are in the url the Kemal server picks them up instead of the Mint app - In Elm it uses location hash for routing. Could we do the same thing - and optionally choose to use location hash as I think this would solve my issues running inside the same server.

**Comment by kingsleyh** [comment]
I've found a workaround in Kemal which seems to work ok

```
  error 404 do |context|
    context.response.content_type = "text/html"
    context.response.status_code = 200
    File.read(File.join("public", "index.html"))
  end
```

it feels a bit wrong to use the error stuff though - it would be better if a custom handler or a middleware could do it instead

**Comment by gdotdesign** [comment]
Would it be okay if this would be supported?

```
routes {
  /path# {
  }

  /path#:entity/:page (entity : String, page : String) {
  }
}
```

**Comment by kingsleyh** [comment]
I think kemal would still pick up the /path part first but I'm not sure - If it's in master I could try and see if it works

**Comment by gdotdesign** [comment]
Should be fixed in 0.4.0 @kingsleyh can you validate it? Should I close this issue?

**Comment by kingsleyh** [comment]
yup all good :)
